File: SleepIn.java

public class SleepIn {

	public static void main(String[] args) {

		System.out.println(sleepIn(false, false)); // true
		System.out.println(sleepIn(true, false)); // false
		System.out.println(sleepIn(false, true)); // true
		 
		 
		 
	} // end main method
	
	
//	The parameter weekday is true if it is a weekday, and the parameter vacation is true if we are on vacation. 
//	We sleep in if it is not a weekday or we're on vacation. 
//	Return true if we sleep in.

	
	public static boolean sleepIn(boolean weekday, boolean vacation) {
		
		if (!weekday && !vacation) {
			return true;
		} else if (!weekday && vacation) {
			return true;
		}
		
		if (weekday) {
			if (vacation) {
				return true;
			} else if(!vacation) {
				return false;
			} else {
				return false;
			}
		}
		return false;
		  
	} // end sleepin method

} // end sleepin class

File: Python/sleep_in.py
def sleep_in(weekday, vacation):
    if vacation == True:
        return True
    elif weekday == False and vacation == True:
        return True
    elif weekday == False and vacation == False:
        return True
    else:
        return False


if __name__ == "__main__":

    print(sleep_in(False, False))  # True
    print(sleep_in(True, False))  # False
    print(sleep_in(False, True))  # True
